lib/google_auth.rb

```diff
@@ -0,0 +1,43 @@
+require 'googleauth'
+require 'google/apis/gmail_v1'
+
+class GoogleAuth
+  def self.service
+    service = Google::Apis::GmailV1::GmailService.new
+    service.client_options.application_name = "GOV.UK Email monitoring service"
+    service.authorization = get_credentials
+    service
+  end
+
+  def self.get_credentials
+    client_id = Google::Auth::ClientId.from_hash({
+      "installed" => {
+        "client_id" => ENV.fetch('GOOGLE_CLIENT_ID'),
+        "client_secret" => ENV.fetch('GOOGLE_CLIENT_SECRET'),
+      }
+    })
+
+    authorizer = Google::Auth::UserAuthorizer.new(
+      client_id,
+      Google::Apis::GmailV1::AUTH_GMAIL_READONLY,
+      EnvironmentTokenStore.new
+    )
+
+    authorizer.get_credentials('default')
+  end
+
+  class EnvironmentTokenStore
```

Comment:
A little nitpicky, but perhaps `DummyEnvironmentTokenStore` might be better given how little this object does.